### PR TITLE
fix: access __class__.__name__

### PR DESCRIPTION
**Pull request by NiklasMerz**
Without this fix I get an error that there is no property `__name__`.

### CODE REVIEW

**Comment by leandrodesouzadev** [comment]
Can you also provide the classes involved on the issue, and the traceback?

**Comment by NiklasMerz** [comment]
I am using the `ApmMixin` in my class based views and get this error:

```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/import/list

Django Version: 4.1.2
Python Version: 3.10.6
Installed Applications:
['django.contrib.admin',
 'django.contrib.sites',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django_celery_beat',
 'rest_framework',
 'django_prometheus',
 'django_extensions',
 'frontend',
 'importer',
 'optimizer',
 'repository',
 'jquery_ui',
 'djapm.apm.apps.ApmConfig']
Installed Middleware:
['django_prometheus.middleware.PrometheusBeforeMiddleware',
 'djapm.apm.middlewares.ApmMetricsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django_prometheus.middleware.PrometheusAfterMiddleware',
 'djapm.apm.middlewares.ErrorTraceMiddleware']



Traceback (most recent call last):
  File "project-path/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "project-path/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "project-path/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "project-path/venv/lib/python3.10/site-packages/djapm/apm/views.py", line 28, in dispatch
    contrib._contribute_to_request(request, view=self, logger_name=self.logger_name)
  File "project-path/venv/lib/python3.10/site-packages/djapm/apm/contrib.py", line 25, in _contribute_to_request
    view_name = ".".join([view.__module__, view.__name__])

Exception Type: AttributeError at /import/list
Exception Value: 'ProjectListView' object has no attribute '__name__'

```

File: djapm/apm/contrib.py [hunk 1]
```diff
@@ -19,10 +19,10 @@ def _contribute_to_request(
         data = rest_request.data
         view_name = get_view_name(view)
         if view_name == "Function":
-            view_name = ".".join([view.__module__, view.__name__])
+            view_name = ".".join([view.__module__, view.__class__.__name__])
```
Comment (leandrodesouzadev) [inline djapm/apm/contrib.py]: I think that i prefer this:
`view_name = ".".join([view.__module__, getattr(view, "__name__", view.__class__.__name__]))`

**Review by leandrodesouzadev — COMMENTED** [review]
Let me know what you think.

File: djapm/apm/contrib.py (hunk 1 above)
Comment (NiklasMerz) [reply djapm/apm/contrib.py]: I updated the branch with this code and it works. So it's good for me.

**Comment by leandrodesouzadev** [comment]
Hey @NiklasMerz i'm trying to implement a test case here to this case, but i'm ain't able to reproduce the issue you're pointing.
The steps i'm doing:

```python
# example/polls/views.py

class PollsView(ListView, ApmView):
    queryset = Poll.objects.all()


# example/polls/urls.py
# Added a new url pattern for the CBV
urlpatterns = [
  # ...
  path("polls-cbv/", views.PollsView.as_view(), name="polls-page-list-cbv"),
  # ...

# tests/test_apm/test_contrib.py
from django.test.client import RequestFactory
from django.urls import reverse

from djapm.apm.contrib import _contribute_to_request
from polls.views import PollsView


def test_contribute_to_request_on_class_based_views(rf: RequestFactory):
    request = rf.get(reverse("polls-page-list-cbv"))
    _contribute_to_request(request, view=PollsView.as_view(), logger_name=None)
```

The above test does not raise any exceptions, i had tested this with python3.9 on django4.1.1 and django4.1.2

**Comment by NiklasMerz** [comment]
I definitely get the error `'ProjectListView' object has no attribute '__name__'` without this fix. My class looks like this:

```python
class ProjectListView(ListView, ApmView):
    model = Project
    template_name = "importer/project_list.html"
    context_object_name = "projects"

    def get_queryset(self):
        return Project.objects.all().order_by("-created")
```

**Comment by leandrodesouzadev** [comment]
Can i also see your `path` entry for this view on your `urls.py`?

**Comment by NiklasMerz** [comment]
Sure 

`path("list", ProjectListView.as_view(), name="project_list"),`

**Comment by leandrodesouzadev** [comment]
Hmm, that looks right.
I'm also not really happy with the current implementation for the `view_name` thing.
I think that i will give some love to that method, and make sure that this gets corrected.
For now i will close this PR and open an issue on this to do some refactoring.

**Comment by NiklasMerz** [comment]
I am not entirely sure what is different in my project. Sounds good to me. Thanks for taking care of this.

**Pull request closed without merging**